src/transformers/models/siglip/modeling_siglip.py

```diff
@@ -926,7 +926,7 @@ def __init__(self, config: SiglipTextConfig):
         self.embeddings = SiglipTextEmbeddings(config)
         self.encoder = SiglipEncoder(config)
         self.final_layer_norm = nn.LayerNorm(embed_dim, eps=config.layer_norm_eps)
-
+        #if not (config.vision_config.image_size==256 and config.text_config.vocab_size==250000 and config.vision_config.patch_size==16):
```

Comment:
@NielsRogge we cannot check the vision config here because `TextTransformer` is constructed from text config. We need to add an attr to the config otherwise there's no other way of checking this.

These are the specs of this new model BTW that's why I'm specifically checking them, but I find it a bit hacky and it won't be possible unless I change it to take in complete config but this is against the patterns so I wouldn't like to do that.